src/source/native.rs

```diff
@@ -70,122 +76,206 @@ impl NativeServer {
 
 fn handle_tcp(
     chans: util::Channel,
-    tags: metric::TagMap,
-    listner: TcpListener,
-) -> thread::JoinHandle<()> {
-    thread::spawn(move || {
-        for stream in listner.incoming() {
-            if let Ok(stream) = stream {
-                debug!(
-                    "new peer at {:?} | local addr for peer {:?}",
-                    stream.peer_addr(),
-                    stream.local_addr()
-                );
-                let tags = tags.clone();
-                let chans = chans.clone();
-                thread::spawn(move || {
-                    handle_stream(chans, tags, stream);
+    tags: sync::Arc<metric::TagMap>,
+    listener_map: HashMap<mio::Token, mio::net::TcpListener>,
```

Comment:
Marker for `HashMap` -> slab conversion.

Reply:
Fixed.